src/coordination/coordinator_instance.cpp

```diff
@@ -321,15 +385,14 @@ auto CoordinatorInstance::RegisterReplicationInstance(CoordinatorToReplicaConfig
 
   if (!new_instance->SendDemoteToReplicaRpc()) {
     spdlog::error("Failed to send demote to replica rpc for instance {}", config.instance_name);
+    // TODO(antoniofilipovic) Do we need this here
     undo_action_();
     return RegisterInstanceCoordinatorStatus::RPC_FAILED;
   }
 
   if (!raft_state_.AppendRegisterReplicationInstanceLog(config)) {
-    undo_action_();
```

Comment:
Why deleted, I think we need this?

Reply:
We are not undoing anything in this part, everything is done in force reset, so I would say we don't need this